src/components/modules/ui.ts

```diff
@@ -501,6 +505,29 @@ export default class UI extends Module<UINodes> {
     this.Editor.Toolbar.close();
   }
 
+  /**
+   * @param {ClipboardEvent} event - keyboard event
+   */
+  private async handlePasteAnyBlockSelected(event: ClipboardEvent): Promise<void> {
```

Comment:
We have a Paste module that is responsible for pasting logic. The `ui` module is not the best place for it